idseq_pipeline/commands/host_filtering_functions.py

```diff
@@ -227,12 +227,17 @@ def run_star_part(output_dir, genome_dir, fastq_files, count_genes=False):
                            '--genomeDir', genome_dir,
                            '--readFilesIn', " ".join(fastq_files)]
     if fastq_files[0][-3:] == '.gz':
-        star_command_params += ['--readFilesCommand', 'zcat']
+        if "fasta" in fastq_files[0]:
+            max_lines = MAX_INPUT_READS * 2
+        else:
+            max_lines = MAX_INPUT_READS * 4
+        # create a custom decompressor which does "zcat $input_file | head - ${max_lines}"
+        execute_command("echo 'zcat ${2} | head -${1}' > %s/gzhead; " % genome_dir)
```

Comment:
sorry i don't understand where ${2} and ${1} are coming from - are those in the bash environment?

Reply:
it's generating a shell script. ${2} is supplied by star as the input file. ${1} is the number of max lines. See documentation above the line.

Reply:
ah i see. would it make sense to just check in gzhead somewhere and call it rather than recreating it each time?